turbopack/crates/turbo-tasks-backend/src/kv_backing_storage.rs

```diff
@@ -323,16 +340,34 @@ impl<T: KeyValueDatabase + Send + Sync + 'static> BackingStorage
                     for (task_id, meta, data) in task_items_result?.into_iter().flatten() {
                         let key = IntKey::new(*task_id);
                         let key = key.as_ref();
-                        if let Some(meta) = meta {
+                        if let Some((meta, rcstr_map)) = meta {
+                            let global_ids = save_strings_serial(batch, &rcstr_map)?;
+                            let mut meta_bytes = SmallVec::<[u8; 16]>::new();
+                            global_ids.write_to(&mut meta_bytes)?;
+                            meta_bytes.extend_from_slice(&meta);
```

Comment:
Looks like this code is repeated a bunch of times. I guess you want to have a helper for that to construct `final_bytes` from `batch` + `seralized_bytes` + `rcstr_map`